api-registration/server/index.js

```diff
@@ -0,0 +1,64 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+
+  /**
+   * Hash the user's password with `argon2.hash()`
+   * Then, 😉
+   *   Insert the user's "username" and "hashedPassword" into the "users" table.
+   *   Then, 😉
+   *     Respond to the client with a 201 status code and the new user's "userId", "username", and "createdAt" timestamp.
+   *   Catch any errors.
+   * Catch any errors.
+   *
+   * Hint: Insert statements can include a `returning` clause to retrieve the insterted row(s).
+   */
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+        insert into "users" ("username", "hashedPassword")
+        values ($1, $2)
+        returning "userId", "username", "createdAt"
+      `;
+      const params = [username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          const newUser = result.rows;
+          res.status(201).json(newUser);
```

Comment:
This will return an array with the one user inside of it, but we want to send back just the user object. So on line 51 you want to use array destructuring or directly access index 0.

Destructuring:
```JAVASCRIPT
const [ newUser ] = result.rows;
```

OR Accessing index 0
```JAVASCRIPT
const newUser = result.rows[0];
```